roxie/ccd/ccdqueue.cpp

```diff
@@ -2114,40 +2115,44 @@ class DelayedPacketQueueChannel : public CInterface
     inline unsigned queryChannel() const { return channel; }
     inline DelayedPacketQueue &queryQueue(unsigned subchannel)
     {
-        if (subchannel > active)
-            active = subchannel;
+        assertex(subchannel);  // Subchannel 0 means primary and is never delayed
+        subchannel -= 1;
+        if (subchannel > maxSeen)
+            maxSeen = subchannel;
         return queues[subchannel];
     }
-    unsigned timeout() const
+    unsigned timeout(unsigned now) const
     {
         unsigned min = (unsigned) -1;
-        for (unsigned queue = 0; queue < active; queue++)
+        for (unsigned queue = 0; queue <= maxSeen; queue++)
         {
-            unsigned t = queues[queue].timeout();
+            unsigned t = queues[queue].timeout(now);
             if (t < min)
                 min = t;
         }
         return min;
     }
     void checkExpired(
+            unsigned now,
 #ifdef ROXIE_SLA_LOGIC
             RoxieQueue &slaQueue,
 #endif
             RoxieQueue &hiQueue, RoxieQueue &loQueue)
     {
-        for (unsigned queue = 0; queue < active; queue++)
+        for (unsigned queue = 0; queue <= maxSeen; queue++)
```

Comment:
does it matter that this will call checkExpired in queue 0, even if it hasn't been seen?

Reply:
No, an empty queue is fine.